crates/primitives/src/transaction/mod.rs

```diff
@@ -305,50 +305,33 @@ impl Transaction {
         }
     }
 
-    // TODO: dedup with effective_tip_per_gas
-    /// Determine the effective gas limit for the given transaction and base fee.
-    /// If the base fee is `None`, the `max_priority_fee_per_gas`, or gas price for non-EIP1559
-    /// transactions is returned.
-    ///
-    /// If the `max_fee_per_gas` is less than the base fee, `None` returned.
-    pub fn effective_gas_tip(&self, base_fee: Option<u64>) -> Option<u128> {
-        if let Some(base_fee) = base_fee {
-            let max_fee_per_gas = self.max_fee_per_gas();
-
-            if max_fee_per_gas < base_fee as u128 {
-                None
-            } else {
-                let effective_max_fee = max_fee_per_gas - base_fee as u128;
-                Some(std::cmp::min(effective_max_fee, self.priority_fee_or_price()))
-            }
-        } else {
-            Some(self.priority_fee_or_price())
-        }
-    }
-
     /// Returns the effective miner gas tip cap (`gasTipCap`) for the given base fee:
     /// `min(maxFeePerGas - baseFee, maxPriorityFeePerGas)`
     ///
+    /// If the base fee is `None`, the `max_priority_fee_per_gas`, or gas price for non-EIP1559
+    /// transactions is returned.
+    ///
     /// Returns `None` if the basefee is higher than the [Transaction::max_fee_per_gas].
-    pub fn effective_tip_per_gas(&self, base_fee: u64) -> Option<u128> {
-        let base_fee = base_fee as u128;
+    pub fn effective_tip_per_gas(&self, base_fee: Option<u64>) -> Option<u128> {
+        // Convert the base fee to u128 if present, or provide a default value
+        let base_fee = base_fee.unwrap_or_default() as u128;
```

Comment:
sorry, actually to behave like the old method, if `base_fee == None`, then this needs to return:
* the priority fee, if it exists, or
* the gas price, if the priority fee does not

Reply:
> sorry, actually to behave like the old method, if `base_fee == None`, then this needs to return:
> 
> * the priority fee, if it exists, or
> * the gas price, if the priority fee does not

alright so if i understand well, i need to add these right ? 

```rust
  if base_fee.is_none() {
        return self.max_priority_fee_per_gas().or_else(|| self.priority_fee_or_price());
    }
```

Reply:
This would be using _just_ `priority_fee_or_price`:
```rust
if base_fee.is_none() {
    return self.priority_fee_or_price()
}